source/custom/nixnet_service.custom.cpp

```diff
@@ -24,31 +24,35 @@ inline bool status_ok(int32 status)
 }
 
 // Helper to read fields of u32 returned by nxReadState of nxState_CANComm and set them on gRPC message
-void SetCanComm(const u32& input, nixnet_grpc::CanComm* output)
+void SetCanCommResponse(const u32& input, nixnet_grpc::CanCommResponse* output)
 {
-  output->set_comm_state(nxCANComm_Get_CommState(input));
+  output->set_comm_state(static_cast<nixnet_grpc::CanCommState>(input));
```

Comment:
This is wrong. You still need to call `nxCANComm_Get_CommState` to get the value representing `comm state`, then typecast it to enum type. Same applies for all response enum changes in this file. You can probably store value returned by `nxCANComm_Get_CommState` in a variable and use that to set enum and raw values.

Reply:
Fixed this.